tools/llvm-kompile-compute-ordinal/main.cpp

```diff
@@ -0,0 +1,99 @@
+#include <kllvm/ast/AST.h>
+#include <kllvm/parser/KOREParser.h>
+
+#include <llvm/Support/CommandLine.h>
+
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+using namespace llvm;
+using namespace kllvm;
+
+cl::OptionCategory ordinal_cat("llvm-kompile-compute-ordinal options");
+
+cl::opt<std::string> kompiled_dir(
+    cl::Positional, cl::desc("<kompiled-dir>"), cl::Required,
+    cl::cat(ordinal_cat));
+
+cl::opt<std::string> line(
+    cl::Positional, cl::desc("<line>"), cl::Required, cl::cat(ordinal_cat));
+
+cl::opt<bool> is_k_line(
+    cl::Positional, cl::desc("[is-k-line]"), cl::init(false),
+    cl::cat(ordinal_cat));
+
+int64_t get_location(kore_axiom_declaration *axiom) {
```

Comment:
This function is a duplicate of the function with the same name in the `llvm-kompile-compute-loc` tool. And more generally, there seems to exist a lot of duplicate code between the two tools. I am wondering if it is worth it to refactor the code in a way that avoids duplication. Maybe add some functions with the common functionality in `lib/codegen/Util.cpp`?

Reply:
I'm not sure if putting something only used by tools in the `codegen` directory is the appropriate strategy here. I would like to export it somewhere to reuse, but I am just not sure if this is the right place.

Reply:
maybe @Baltoli and/or @dwightguth could offer a better suggestion

Reply:
I think this shared code could probably just live in a new module of the AST library?

Reply:
Sure, but why not in a new header module in `kllvm/util`?

Reply:
`trim` is also shared between the two tools. Should I put in this new module as well?

Reply:
Yes, anything you've shared between these tools should be moved to the new module.

Reply:
It should be in `kllvm/util` or `kllvm/ast`?

Reply:
I think `ast` is fine here; these are all functions that operate on or are related to the AST data structures.

Reply:
Perfect! Thanks!